crates/hotshot/task-impls/src/quorum_proposal/handlers.rs

```diff
@@ -501,39 +501,20 @@ impl<TYPES: NodeType, V: Versions> ProposalDependencyHandle<TYPES, V> {
                 );
             }
         }
-        let block_header = if version < V::Marketplace::VERSION {
-            TYPES::BlockHeader::new_legacy(
-                state.as_ref(),
-                self.instance_state.as_ref(),
-                &parent_leaf,
-                commitment_and_metadata.commitment,
-                builder_commitment,
-                metadata,
-                commitment_and_metadata.fees.first().clone(),
-                version,
-                *self.view_number,
-            )
-            .await
-            .wrap()
-            .context(warn!("Failed to construct legacy block header"))?
-        } else {
-            TYPES::BlockHeader::new_marketplace(
-                state.as_ref(),
-                self.instance_state.as_ref(),
-                &parent_leaf,
-                commitment_and_metadata.commitment,
-                commitment_and_metadata.builder_commitment,
-                commitment_and_metadata.metadata,
-                commitment_and_metadata.fees.to_vec(),
-                *self.view_number,
-                commitment_and_metadata.auction_result,
-                version,
-            )
-            .await
-            .wrap()
-            .context(warn!("Failed to construct marketplace block header"))?
-        };
-
+        let block_header = TYPES::BlockHeader::new_legacy(
```

Comment:
should new_legacy() be renamed to new() ?

Reply:
yeah sure, a lot is missing from this PR still :D